src/V3Timing.cpp

```diff
@@ -0,0 +1,785 @@
+                                                     new AstVarRef{flp, newvscp, VAccess::READ}}};
+        ifp->addIfsp(new AstAssign{flp, new AstVarRef{flp, chgVscp, VAccess::WRITE},
+                                   new AstConst{flp, AstConst::BitFalse()}});
+        auto* const alwaysp = new AstAlwaysPost{
+            flp,
+            new AstSenTree{flp, new AstSenItem{flp, VEdgeType::ET_CHANGED,
+                                               new AstVarRef{flp, chgVscp, VAccess::READ}}},
+            ifp};
+        m_procp->addNext(alwaysp);
+        // Signal that we are doing this assignment
+        auto* const enablep = new AstAssign{flp, new AstVarRef{flp, chgVscp, VAccess::WRITE},
+                                            new AstConst{flp, AstConst::BitTrue()}};
+        nodep->replaceWith(enablep);
+        // Assign to the temporary var
+        auto* const assignp = new AstAssign{flp, new AstVarRef{flp, newvscp, VAccess::WRITE},
+                                            nodep->rhsp()->unlinkFrBack()};
+        if (controlp) {
+            if (m_underFork) {
+                controlp->replaceWith(new AstBegin{flp, "", assignp});
+                assignp->addNextHere(controlp);  // No need to create another fork
+            } else {
+                controlp->replaceWith(assignp);
+                auto* const forkp = new AstFork{flp, "", controlp};
+                forkp->joinType(VJoinType::JOIN_NONE);
+                assignp->addNextHere(forkp);
+            }
+        } else {
+            enablep->addNextHere(assignp);
+        }
+        nodep->deleteTree();
+    }
+    virtual void visit(AstWait* nodep) override {
+        // Gather all vars we want to wait on
+        VarScopeSet waitVars;
+        nodep->foreach<AstVarRef>([&](AstVarRef* refp) { waitVars.insert(refp->varScopep()); });
+        if (waitVars.empty()) {  // There are no vars to wait on
+            if (nodep->bodysp())
+                nodep->replaceWith(nodep->bodysp()->unlinkFrBackWithNext());
+            else
+                nodep->unlinkFrBack();
+        } else {
+            FileLine* const flp = nodep->fileline();
+            AstNode* senitemsp = nullptr;
+            // Wait on changed events related to the vars in the wait statement
+            for (auto* const vscp : waitVars) {
+                const bool isEvent = vscp->dtypep() && vscp->dtypep()->basicp()
+                                     && vscp->dtypep()->basicp()->isEvent();
+                const auto edgeType = isEvent ? VEdgeType::ET_EVENT : VEdgeType::ET_CHANGED;
+                senitemsp = AstNode::addNext(
+                    senitemsp,
+                    new AstSenItem{flp, edgeType, new AstVarRef{flp, vscp, VAccess::READ}});
+            }
+            auto* const condp = nodep->condp()->unlinkFrBack();
+            auto* const eventControlp = new AstEventControl{
+                flp, new AstSenTree{flp, VN_AS(senitemsp, SenItem)}, nullptr};
```

Comment:
You should run `V3Const::constifyExpensiveEdit` on the SenTree, which will sort the terms so you have fewer unique sen trees. (This also removes dupliate terms, so you need not necessarily need to pick up only the unique vars)

Reply:
Fixed